sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveHints.scala

```diff
@@ -50,11 +51,15 @@ object ResolveHints {
    *
    * This rule must happen before common table expressions.
    */
-  class ResolveJoinStrategyHints(conf: SQLConf) extends Rule[LogicalPlan] {
+  class ResolveJoinStrategyHints(
+      conf: SQLConf,
+      catalogLookup: String => CatalogPlugin) extends Rule[LogicalPlan] with LookupCatalog {
```

Comment:
I think it is cleaner to pass an existing `LookupCatalog` instance. In the v2 session catalog PR, I updated `DataSourceResolution` to use a lookup that is passed in instead: https://github.com/apache/spark/pull/24768/files#diff-acba0f0ea96f6511882f4ead4fb33dc2R44

I think that would be a cleaner way to get the identifier extractors.

Reply:
Not used. Removed.

Reply:
Like suggested approach of passing LookupCatalog instance because it will make it easier to add feature to LookCatalog e.g. default catalog name.